tests/src/testadaptivestepsizing.cpp

```diff
@@ -0,0 +1,229 @@
+  std::vector<ElementType> fes;
+
+  for (auto& element : elements(gridView))
+    fes.emplace_back(basis, element, E, nu, thickness, LoadDefault(), &neumannBoundary, neumannBoundaryLoad);
+
+  auto basisP = std::make_shared<const decltype(basis)>(basis);
+  Ikarus::DirichletValues dirichletValues(basisP->flat());
+
+  dirichletValues.fixDOFs([&](auto& basis, auto&& dirichletFlags) {
+    Dune::Functions::forEachBoundaryDOF(basis, [&](auto&& localIndex, auto&& localView, auto&& intersection) {
+      if (std::abs(intersection.geometry().center()[0] + 0.5) < 1e-8)
+        dirichletFlags[localView.index(localIndex)] = true;
+    });
+  });
+
+  dirichletValues.fixDOFs([&](auto& basis, auto&& dirichletFlags) {
+    for (auto fixedDirection = 1; fixedDirection < 3; ++fixedDirection) {
+      Dune::Functions::forEachBoundaryDOF(Dune::Functions::subspaceBasis(basis, fixedDirection),
+                                          [&](auto&& localIndex, auto&& localView, auto&& intersection) {
+                                            if (std::abs(intersection.geometry().center()[0] - 0.5) < 1e-8)
+                                              dirichletFlags[localView.index(localIndex)] = true;
+                                          });
+    }
+  });
+
+  auto sparseAssembler = SparseFlatAssembler(fes, dirichletValues);
+
+  Eigen::VectorXd d;
+  d.setZero(basis.flat().size());
+  double lambda = 0.0;
+
+  auto req = FErequirements().addAffordance(Ikarus::AffordanceCollections::elastoStatics);
+
+  auto residualFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
+    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
+        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
+    return sparseAssembler.getVector(req);
+  };
+
+  auto KFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
+    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
+        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
+    return sparseAssembler.getMatrix(req);
+  };
+
+  auto nonLinOp  = Ikarus::NonLinearOperator(functions(residualFunction, KFunction), parameter(d, lambda));
+  auto linSolver = Ikarus::LinearSolver(Ikarus::SolverTypeTag::sd_SimplicialLDLT);
+
+  int loadSteps = 6;
+
+  auto nr               = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
+  auto dass             = Ikarus::DefaultAdaptiveStepSizing{};
+  dass.targetIterations = targetIterations;
+
+  /// control routine with and without step sizing
+  auto crWSS  = Ikarus::PathFollowing(nr, loadSteps, stepSize, pft, dass);
+  auto crWoSS = Ikarus::PathFollowing(nr, loadSteps, stepSize, pft);
+
+  auto nonLinearSolverObserver = std::make_shared<Ikarus::NonLinearSolverLogger>();
+  auto pathFollowingObserver   = std::make_shared<Ikarus::ControlLogger>();
+  nr->subscribeAll(nonLinearSolverObserver);
+
+  try {
```

Comment:
You should use `t.checkThrow` and don't write the try and catch yourself. Also `Dune::IOError` is the wrong exception thrown in the observer can you replace this with something more reasonable.

Full disclosure: I added   `t.checkThrow`  to dune-common, since I was tired of writing this :)

Reply:
done